client/hwlib/src/helpers.rs

```diff
@@ -28,17 +28,52 @@ pub(crate) fn append_to_pathbuf(p: PathBuf, s: &str) -> PathBuf {
 
 #[cfg(test)]
 pub(crate) mod test_utils {
+    use crate::collectors::os_info::CommandRunner;
     use anyhow::{bail, Result};
-    use std::{env, fs::read_dir, path::PathBuf};
+    use std::{
+        collections::HashMap,
+        env,
+        fs::{read_dir, read_to_string},
+        path::{Path, PathBuf},
+    };
+
+    type SystemCommand<'args> = (&'args str, Vec<&'args str>);
+
+    pub(crate) struct MockCommandRunner<'args> {
+        calls: HashMap<SystemCommand<'args>, Result<&'args str>>,
+    }
+
+    impl<'args> MockCommandRunner<'args> {
+        pub(crate) fn new(calls: Vec<(SystemCommand<'args>, Result<&'args str>)>) -> Self {
+            let calls = calls
+                .into_iter()
+                .map(|((cmd, args), res)| ((cmd, args.to_vec()), res))
```

Comment:
You don't need this map here as `args` is already a Vec and if you remove `.to_vec()` this is just identity :slightly_smiling_face: